pkg/build/pipelines/git-checkout.yaml

```diff
@@ -46,9 +46,43 @@ pipeline:
       fail() { msg FAIL "$@"; exit 1; }
       vr() { msg "execute:" "$@"; "$@"; }
 
+      process_cherry_picks() {
+        local cpicksf="$1" oifs="" count=0
+        if [ -z "$cpicksf" ]; then
+            return 0
+        fi
+        if [ ! -f "$cpicksf" ]; then
+            msg "cherry picks input '$cpicksf' is not a file"
+            return 1
+        fi
+
+        local line="" branch="" hash=""
+        while read line; do
+            # drop anything after #
+            line=${line%%#*}
+            [ -z "$line" ] && continue
+            case "$line" in
+              */*) branch=${line#*/}; hash=${line%%/*};;
+              *) hash=${line};;
+            esac
+            if [ -n "$branch" ]; then
+                # if you haven't already grabbed that branch, 
+                # grab that branch with git fetch
+                vr git fetch origin $branch:$branch
```

Comment:
you have to return failure if this fails.

Reply:
and ideally we don't 'git fetch main' N times for each cherry-pick.  so if we've seen the same branch we can skip.